packages/jest-reporters/src/GithubActionsLogsReporter.ts

```diff
@@ -0,0 +1,366 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {ProjectConfig} from '@jest/types/src/Config';
+import chalk = require('chalk');
+import type {
+  AggregatedResult,
+  AssertionResult,
+  Test,
+  TestContext,
+  TestResult,
+} from '@jest/test-result';
+import DefaultReporter from './DefaultReporter';
+import type {ReporterOnStartOptions} from './types';
+
+type performaceInfo = {
+  end: number;
+  runtime: number;
+  slow: boolean;
+  start: number;
+};
+
+type resultTreeLeaf = {
+  name: string;
+  passed: boolean;
+  duration: number;
+  children: Array<never>;
+};
+
+type resultTreeNode = {
+  name: string;
+  passed: boolean;
+  children: Array<resultTreeNode | resultTreeLeaf>;
+};
+
+type resultTree = {
+  children: Array<resultTreeLeaf | resultTreeNode>;
+  name: string;
+  passed: boolean;
+  performanceInfo: performaceInfo;
+};
+
+export default class GithubActionsLogsReporter extends DefaultReporter {
```

Comment:
any particular reason why this is separate from our existing GH actions reporter?

Reply:
That is because you Github Actions reporter is used only when requested by the user (by adding 'github-actions' to the reporters) and running in a GITHUB ACTIONS environment, while (as I understood your request) this new reporter should be the new default if used in a GITHUB ACTIONS environment.

Reply:
The I didn't word myself properly - I'd like for this new capability to be part of our existing actions reporter